ols/app/endpoints/ols.py

```diff
@@ -222,13 +219,23 @@ def generate_response(
 def store_conversation_history(
     user_id: str, conversation_id: str, llm_request: LLMRequest, response: Optional[str]
 ) -> None:
-    """Store conversation history into selected cache."""
+    """Store conversation history into selected cache.
+
+    History is stored as simple dictionaries in the following format:
+    ```python
+        [
+            {"type": "human/ai", "content"},
+            ...
+        ]
+    ```
+    """
     try:
         if config.conversation_cache is not None:
             logger.info(f"{conversation_id} Storing conversation history.")
-            chat_message_history = ChatHistory.get_chat_message_history(
-                llm_request.query, response or ""
-            )
+            chat_message_history = [
+                {"type": "human", "content": llm_request.query},
```

Comment:
I'd probably add "type", "human" etc. into constants (to avoid typos) or rather have two constructors prepared (that simply return dicts)